### ISSUE TITLE
worms gaps and switching

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#1844]
@sharpphyl attached is names used by DMNS:Inv and not in WoRMS.

110 of them have not_exists in classification_id - those have nothing in the Arctos classification to copy over. I'm not sure if you want to do something with those before I proceed or not.

I need to 

1) push these classifications to "WoRMS (via Arctos)"
2) Switch your collection over to use the "WoRMS (via Arctos)"

Let me know when you're ready for me to do either or both of those things.

[temp_now_w_dmns_i.csv.zip](https://github.com/ArctosDB/arctos/files/2665338/temp_now_w_dmns_i.csv.zip)

**Comment by sharpphyl** [comment]
I'm confused.  I checked about five of these and all of them have an Arctos classification.  Here's one.

![screen shot 2018-12-10 at 8 19 11 pm](https://user-images.githubusercontent.com/15368365/49776357-f0dd0a00-fcb8-11e8-8cae-eb486046a189.png)

I must be misinterpreting your posting.  Can you clarify what I need to do?  I know that WoRMS doesn't have these taxa (they're mostly terrestrial species or legacy taxa).  But so far everything has an Arctos classification. I'll keep checking.

**Comment by campmlc** [comment]
Phyllis, it looks like there are 110 out of the 1780 taxa that need
classifications added - they have "not exists' in the classification ID.

**Comment by dustymc** [comment]
Yes it's just 110. No you don't _need_ to do anything. You can, if you wish, add some data to the Arctos classification before I pull this, or to the worms after, or do nothing, or whatever.

**Comment by sharpphyl** [comment]
I've started fixing them, but if I can add data later you pull WoRMs that would be good as it will take a while to do all 110.

**Comment by dustymc** [comment]
These are moved. You can update the WoRMS (via Arctos) classification  - preferably by just adding an aphiaid and clicking refresh if they get these taxa.

**Comment by sharpphyl** [comment]
Got it.

And it's ok to switch our source from Arctos to WoRMS (via Arctos)??

**Comment by dustymc** [comment]
Let's find out....you're switched.

**Comment by sharpphyl** [comment]
Ok, we'll scream if anything weird happens!

**Comment by dustymc** [comment]
Looks like the refresh is already caught up, so scream away!

**Comment by sharpphyl** [comment]
Well, it doesn't look like everything made it into our upload.  The first thing I found I needed to add was _Bellamya unicolor_.

![screen shot 2018-12-12 at 7 51 56 pm](https://user-images.githubusercontent.com/15368365/49912473-87d7cc80-fe47-11e8-8f3c-21d4b0ed82fc.png)


But there's no WoRMS (via Arctos) entry for it.  

![screen shot 2018-12-12 at 7 52 11 pm](https://user-images.githubusercontent.com/15368365/49912494-93c38e80-fe47-11e8-8c04-6300b7b1a25d.png)

If I create it with the alphiaID will it automatically update?  

Also there should be a long list of this genus, but there are only three (one I created) 

![screen shot 2018-12-12 at 7 54 34 pm](https://user-images.githubusercontent.com/15368365/49912567-cd949500-fe47-11e8-8910-6c1987feb770.png)


![screen shot 2018-12-12 at 7 55 26 pm](https://user-images.githubusercontent.com/15368365/49912599-e56c1900-fe47-11e8-8030-34e439b041a4.png)

Should this have been part of our WoRMS (via Arctos) upload?  And should it automatically become a complete taxon in WoRMS (via Arctos) or do I have to trigger it?

**Comment by dustymc** [comment]
>Bellamya unicolor

That's not in the download I got from them.

```
UAM@ARCTOS> select * from temp_worms2 where scientificName='Bellamya unicolor';

no rows selected

```

I checked the .txt too, it's not there.

<img width="1020" alt="screen shot 2018-12-13 at 7 50 54 am" src="https://user-images.githubusercontent.com/5720791/49950032-ddde5b80-feab-11e8-811d-21e81d992a4d.png">

Wild guess: they hide from downloads the same stuff they hide from "normal" searches. If you can talk them out of those data I can create more stuff....

>create it with the alphiaID 

Create a classification, feed it an aphiaid, refresh (beside aphiaid) - should be Voila! I'm going to try to automate this at some point, but it will be laggy - I think you'll always have the button available for just this sort of use.

**Comment by sharpphyl** [comment]
WoRMS does have two search approaches - the Quick search does not include fossil and non-marine species.  The search at the Taxa menu can be adjusted to include everything, so I don't think they would have any issue with providing all taxa since they're available with an open search.

I assumed they would give you everything - marine, freshwater, fossil, terrestrial, accepted and unaccepted, etc.  Maybe we just need to confirm if they limited what they provided for some reason.  This morning we ran into another one - _Axelella minima_ - which is unaccepted but marine.  Can you tell from the list they gave you if it's only accepted species?  

![axelella minima](https://user-images.githubusercontent.com/15368365/49952490-454ed800-feb9-11e8-8275-84e66b651187.JPG)

I'm not sure who to go to to ask.  Is it best to go to whomever sent you the data?

Where needed, I will create the taxon with the aphidID and refresh to pick it up

**Comment by dustymc** [comment]
>Axelella minima

Thanks!! Yes, a bunch that had no "accepted" variant fell through the cracks. The names are created and I'm making classifications now. These contain some new potential values for taxon_status.

taxon inquirendum: 22
alternate representation:  10772
uncertain: 24

what should I do with those? I can 

* ignore them
* map them to something in http://arctos.database.museum/info/ctDocumentation.cfm?table=CTTAXON_STATUS
* add something to http://arctos.database.museum/info/ctDocumentation.cfm?table=CTTAXON_STATUS and map them to that

**Comment by Jegelewicz** [comment]
From WoRMS:

> Possible statuses for a name:
> 
> Accepted: the used name is accepted in the present literature
> 
> Unaccepted: The used name is NOT accepted in the present literature
> 
> Nomen nudum: a name that does not comply with the name requirements of the codes, such as lack of a description or diagnoses or reference to a description or diagnosis or a type specimen is lacking for publications after 1999
> 
> Alternate representation: to link species that are represented twice: once with and once without subgenus. Alternate representation can also be used for a species and its nominal subspecies (note: you can only add a subspecies if the species is present in the database). See example in the box below
>
> Nomen dubium: a name of uncertain application, because it is not possible to establish the taxon to which it should be referred. A good example is the "Ascothoracida" genus Laocoon. There is a debate whether this is based on a parasite or on a detached piece of the host. It is clearly a dubious name
> 
> Temporary name: to create higher rank taxa to accommodate child taxa for which the classification is not sorted yet
> 
> Taxon inquirendum: an incompletely defined taxon that requires further characterization, it is impossible to identify the taxon
> 
> Interim unpublished: an as yet unavailable name (until in a print issue) which has been published online only, in a work that does not show evidence of ZooBank registration (ICZN Article 8.5)

I think all of these could be good taxon statuses in the code table except Alternate representation, which should have two entries and we should pick either the with or without subgenus classification to use with those.

**Comment by dustymc** [comment]
@sharpphyl I will probably get to this this afternoon. I'll just ignore status if this isn't resolved before then; I think they're mostly fairly obscure taxa and we can pick it up in the refresh if we decide something.

@Jegelewicz if we do something with https://github.com/ArctosDB/arctos/issues/1852 then having multiple classifications under a source will be common. Until then, or until someone who needs an "alternate" is using the "WoRMS (via Arctos)" source, I am adding a single classification to each name under that source.

**Comment by sharpphyl** [comment]
For me, it would be great if we added these to the Taxon status table.  The
alternate representation could go either way.  These tend to be subgenera
that are not the preferred taxon per WoRMS.  As long as you can accept the
subgenus (probably in parentheses) then I'd add that taxon status so that
when we say we uploaded everything from WoRMS, we really did.

**Comment by dustymc** [comment]
> assumed they would give you everything

Me too - they said we'd get everything except algae. I sent them email requesting everything they can give us and they responded with a followup question. If they make anything else available I'll load it.

**Comment by dustymc** [comment]
>subgenera

We accept subgenus, but not in the same way WoRMS does and not as part of a name.

https://github.com/ArctosDB/arctos/issues/1704

>uploaded everything from WoRMS,

We did not. They have a ton of things that do not coincide with our definition of "taxonomy."

>great if we added these

https://github.com/ArctosDB/arctos/issues/1854

**Comment by dustymc** [comment]
>no "accepted" variant fell through the cracks

These should have all been pushed back up through the cracks, which brings the total number of WoRMS (via Arctos) classifications to 554959. Somewhere around 180K of them are new names, if you'd like a scope on the number of buttons you don't have to click. (And I'm a little bummed that this didn't push us over 3M names - we currently have 2839215.)

I'll leave this open until I hear back from WoRMS regarding non-marine and non-extant taxa.

**Comment by sharpphyl** [comment]
All 110 that didn't have a classification are either deleted or now have a classification in WoRMS (via Arctos).  

We did data entry with the new WoRMS yesterday and had no issues.  Once you are able to automate the refresh, it will also show the related taxon for invalid ones.  Right now, only Bryan and I can refresh (manually) but that's working well too.  Two more days of data entry before the holidays so will let you know if we discover anything.  So far the 180k new taxon names are relieving us from having to add any new taxa this week and we couldn't be happier.  Thanks!

**Comment by dustymc** [comment]
Everything you are using in IDs should be refreshed. The rest will take a few weeks.

I just got an email authorizing everything excluded from the original download, so we'll probably be starting over...

>deleted

Why? If they shouldn't be in WoRMS, they'll want to know about them. If there's some remotely defensible reason for WoRMS to have them, they should almost certainly be in Arctos (because people will come from WoRMS and use those names to find stuff in Arctos.)

**Comment by sharpphyl** [comment]
Refresh timing sounds good.  No one will be databasing here again until January 8.

>Why? If they shouldn't be in WoRMS, they'll want to know about them. If there's some remotely defensible reason for WoRMS to have them, they should almost certainly be in Arctos (because people will come from WoRMS and use those names to find stuff in Arctos.)

The ones I deleted were all DMNS:Inv (only) legacy identifications that have been updated to the valid taxon. The reasons included: misspellings that have never been recognized as valid, errors where Arctos is the only thing you find in Google search, and the use of the family as a genus.  None of them were in WoRMS or I would have kept them.  I swallowed hard to create the 32 invalid taxa in WoRMS (via Arctos) but they passed the Google test - in some cases because of the circularity of GBIF picking up our invalid taxa.
